pkg/apis/aws/v1alpha1/account_types.go

```diff
@@ -163,7 +175,7 @@ func (a *Account) IsFailed() bool {
 		string(AccountInternalError),
 	}
 	for _, state := range failedStates {
-		if a.Status.State == state {
+		if string(a.Status.State) == state {
```

Comment:
This is actually a great catch!
Locally I just tried changing this to an array of `AccountStateStatus` instead of strings because those are the valid account `a.Status.State` values, but it can't compile, because here we're checking if the state (of type `AccountStateStatus`) is any failed `AccountConditionType` values.

Reply:
ok, I think we should quit the effort of trying to match AccountConditionType and AccountStateStatus in this PR. It's good that you added a todo, but I guess the PR is already too complex to get another refactor for that. We could create a ticket as a good-first-issue